src/MSBuild/XMake.cs

```diff
@@ -1336,7 +1336,7 @@ private static void ResetGatheringSwitchesState()
                 // This is a hack for now to make sure the perf hit only happens
                 // on diagnostic. This should be changed to pipe it through properly,
                 // perhaps as part of a fuller tracing feature.
-                bool logTaskInputs = verbosity == LoggerVerbosity.Diagnostic;
+                bool logTaskInputs = verbosity == LoggerVerbosity.Diagnostic || isBuildCheckEnabled;
```

Comment:
Note for self: this should ideally be decided on the node after 1st pass ov evaluation and reading of editorconfig. Though it'd as well require the custom analyzers to someho indicate during their registration via the property function what data sources they'll need (to avoid a need to load and register the analyzer in the worker node just for finding it's registrations. And registration in the main node is too late as it happens already after the ProjectEvaluationStartedBuildEventArgs are received - at which point the node can already be running the execution phase)